streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/ForeignJoinSubscriptionSendProcessorSupplier.java

```diff
@@ -77,6 +77,9 @@ public void init(final ProcessorContext context) {
             if (foreignKeySerializer == null) {
                 foreignKeySerializer = (Serializer<KO>) context.keySerde().serializer();
             }
+            if (valueSerializer == null ) {
+                valueSerializer = (Serializer<V>) context.valueSerde().serializer();
+            }
```

Comment:
In addition to the other issues I've pointed out, we also have a few places where we didn't account for nullability in the processors themselves.